### PR TITLE
fix some warning with the use of new crypto functions

### PR DESCRIPTION
**Pull request by hardening**

### CODE REVIEW

File: libfreerdp/crypto/tls.c [hunk 1]
```diff
@@ -1077,14 +1077,15 @@ TlsHandshakeResult freerdp_tls_accept_ex(rdpTls* tls, BIO* underlying, rdpSettin
 		return TLS_HANDSHAKE_ERROR;
 	}
 
-	rdpCertificate* cert = freerdp_settings_get_pointer(settings, FreeRDP_RdpServerCertificate);
+	const rdpCertificate* cert =
```
Comment (akallabeth) [inline libfreerdp/crypto/tls.c:1080]: use `freerdp_settings_get_pointer_writable`

File: libfreerdp/crypto/tls.c [hunk 2]
```diff
@@ -1050,7 +1050,7 @@ TlsHandshakeResult freerdp_tls_accept_ex(rdpTls* tls, BIO* underlying, rdpSettin
 	if (!tls_prepare(tls, underlying, methods, options, FALSE))
 		return TLS_HANDSHAKE_ERROR;
 
-	rdpPrivateKey* key = freerdp_settings_get_pointer(settings, FreeRDP_RdpServerRsaKey);
+	const rdpPrivateKey* key = freerdp_settings_get_pointer(settings, FreeRDP_RdpServerRsaKey);
```
Comment (akallabeth) [inline libfreerdp/crypto/tls.c:1053]: use `freerdp_settings_get_pointer_writable`

File: libfreerdp/crypto/tls.c [hunk 3]
```diff
@@ -1077,14 +1077,15 @@ TlsHandshakeResult freerdp_tls_accept_ex(rdpTls* tls, BIO* underlying, rdpSettin
...
+	const rdpCertificate* cert =
+	    freerdp_settings_get_pointer(settings, FreeRDP_RdpServerCertificate);
 	if (!cert)
 	{
 		WLog_ERR(TAG, "invalid certificate");
 		return TLS_HANDSHAKE_ERROR;
 	}
 
-	status = SSL_use_certificate(tls->ssl, freerdp_certificate_get_x509(cert));
+	status = SSL_use_certificate(tls->ssl, freerdp_certificate_get_x509((rdpCertificate*)cert));
```
Comment (akallabeth) [inline libfreerdp/crypto/tls.c:1088]: not required if `freerdp_settings_get_pointer_writable` is used

**Review by akallabeth — CHANGES_REQUESTED** [review]

File: libfreerdp/crypto/tls.c (hunk 2 above)
Comment (hardening) [reply libfreerdp/crypto/tls.c:1053]: here we don't need to have it writable

**Comment by hardening** [comment]
Fixed changes

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**